src/ServiceMonitor/IISConfigUtil.cpp

```diff
@@ -54,17 +55,16 @@ IISConfigUtil::~IISConfigUtil()
     }
 }
 
-BOOL IISConfigUtil::FilterEnv(unordered_map<wstring, LPTSTR> filter, LPTSTR strEnvName, LPTSTR strEnvValue)
+BOOL IISConfigUtil::FilterEnv(const unordered_map<wstring, LPTSTR>& filter, LPCTSTR strEnvName, LPCTSTR strEnvValue)
 {
     LPTSTR   strFilterValue;
     wstring strFilterName;
     _ASSERT(strEnvName != NULL);
     _ASSERT(strEnvValue != NULL);
 
-    CharUpper(strEnvName);
     strFilterName = strEnvName;

```

Comment:
wstring strFilterName(strEnvName) to have one constructor run. In its current form, you have the default constructor run above, then a copy constructor with line 65.